modules/siddhi-core/src/test/java/org/wso2/siddhi/core/window/DelayWindowTestCase.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.core.window;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.exception.CannotRestoreSiddhiAppStateException;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.query.output.callback.QueryCallback;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.EventPrinter;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+import org.wso2.siddhi.core.util.persistence.InMemoryPersistenceStore;
+import org.wso2.siddhi.core.util.persistence.PersistenceStore;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class DelayWindowTestCase {
+    private static final Logger log = Logger.getLogger(DelayWindowTestCase.class);
+    private boolean eventArrived;
+    private AtomicInteger count = new AtomicInteger(0);
+    private Long lastValue;
+    private double averageValue;
+    private int removeEventCount;
+    private int inEventCount;
+    private boolean error;
+
+    @BeforeMethod
+    public void init() {
+        count.set(0);
+        eventArrived = false;
+        lastValue = Long.valueOf(0);
+        averageValue = Double.valueOf(0);
+        removeEventCount = 0;
+        inEventCount = 0;
+        error = true;
+    }
+
+    @Test
```

Comment:
Can we make use of description annotation of testng to include the description on what we are trying to test in each of these test case?

Reply:
added the description annotation.